daemon/src/maker_cfd_actor.rs

```diff
@@ -300,7 +298,107 @@ impl MakerCfdActor {
+
+        let taker_id = match cfd {
+            Cfd {
+                state: CfdState::IncomingOrderRequest { taker_id, .. },
+                ..
+            } => taker_id,
+            _ => {
+                tracing::warn!("Order is in invalid state. Ignoring trying to accept it.");
+                anyhow::bail!("Order is in invalid state. Ignoring trying to accept it.")
+            }
+        };
+
+        // Update order in db
+        insert_new_cfd_state_by_order_id(
+            cfd.order.id,
+            CfdState::Accepted {
+                common: CfdStateCommon {
+                    transition_timestamp: SystemTime::now(),
+                },
+            },
+            &mut conn,
+        )
+        .await
+        .unwrap();
+
+        self.takers()?
+            .do_send_async(maker_inc_connections_actor::TakerMessage {
+                taker_id,
+                command: TakerCommand::NotifyOrderAccepted { id: msg.order_id },
+            })
+            .await?;
+        self.cfd_feed_actor_inbox
+            .send(load_all_cfds(&mut conn).await?)?;
+
+        self.takers()?
+            .do_send_async(maker_inc_connections_actor::BroadcastOrder(None))
+            .await?;
+        self.order_feed_sender.send(None)?;
+
+        Ok(())
+    }
+
+    async fn handle_reject_order(&mut self, msg: RejectOrder) -> Result<()> {
+        tracing::debug!(%msg.order_id, "Maker rejects an order" );
+
+        let mut conn = self.db.acquire().await?;
+        let cfd = load_cfd_by_order_id(msg.order_id, &mut conn).await?;
+
+        let taker_id = match cfd {
+            Cfd {
+                state: CfdState::IncomingOrderRequest { taker_id, .. },
+                ..
+            } => taker_id,
+            _ => {
+                // TODO: how to tell frontend that this failed?
+                tracing::warn!("Order is in invalid state. Ignoring trying to accept it.");
+                anyhow::bail!("Order is in invalid state. Ignoring trying to accept it.")
+            }
+        };
+
+        // Update order in db
+        insert_new_cfd_state_by_order_id(
+            msg.order_id,
+            CfdState::Rejected {
+                common: CfdStateCommon {
+                    transition_timestamp: SystemTime::now(),
+                },
+            },
+            &mut conn,
+        )
+        .await
+        .unwrap();
+
+        self.takers()?
+            .do_send_async(maker_inc_connections_actor::TakerMessage {
+                taker_id,
+                command: TakerCommand::NotifyOrderRejected { id: msg.order_id },
+            })
+            .await?;
+        self.cfd_feed_actor_inbox
+            .send(load_all_cfds(&mut conn).await?)?;
+
+        // Remove current order
+        self.current_order_id = None;
+        self.takers()?
+            .do_send_async(maker_inc_connections_actor::BroadcastOrder(None))
+            .await?;
+        self.order_feed_sender.send(None)?;
```

Comment:
We could just put it back instead? Otherwise the maker has to create a new one every time they reject an order?

Reply:
This does not seem to work because save the `CFD` in state `rejected` also for the maker.

Reply:
Ah missed that! I guess in some way it is "cleaner" that although it might be a bit annoying.